e2e/src/pages/download.po.ts

```diff
@@ -15,4 +16,12 @@ export class DownloadPage extends AppPage {
   hasFileDownloaded(filePath: string) {
     return fs.existsSync(filePath);
   }
+
+  async clickMoreOptions() {
+    await this.clickElement(by.id('mvMoreOptionsBtn'));
+  }
+
+  async clickDownloadButton() {
```

Comment:
use xpath like can be problematic, because any changes in the document structure would break the test unnecessarily
I would suggest using a class or id to select the element

Reply:
I agree with you. Added generic method that will take care when the object has same reference type.
Same reference has been given  for download, that's causing the issue. id="mvDownloadBtn"